In [1]:
import synapseclient
from synapseclient import Project, File, Folder, Wiki
from synapseclient import Schema, Column, Table, Row, RowSet, as_table_columns
import itertools
import pandas as pd

In [2]:
syn = synapseclient.Synapse()
syn.login()

Welcome, Victor Baham!



### Create a Citations wiki subpage in the ADKP backend portal for each study where its name is the study name in the `Study` column of [Portal - Studies Table](https://www.synapse.org/#!Synapse:syn17083367/tables/)

#### Note: start at index 4 where index i = 0, 1, 2… n because I already manually created Citations wiki subpages for the first 4 studies

In [3]:
query_ps = syn.tableQuery(f"SELECT * FROM {syn.get('syn17083367').id}")
portal_studies = pd.read_csv(query_ps.filepath)

#### Create `Citations` column

In [4]:
portal_studies['Citations'] = ''

#### Note: The name of each Synapse entity in the `Study` column can be accessed with `syn.get('synID').name`

In [5]:
portal_studies

ROW_ID  ROW_VERSION  Study_Type                 Program        Study  \
0        60          599  Consortium              ["AMP-AD"]   syn5550419   
1        61          599  Individual              ["AMP-AD"]  syn16779040   
2        62          599  Individual              ["AMP-AD"]   syn5759376   
3        63          673  Individual  ["AMP-AD", "M20VE-AD"]   syn9705278   
4        64          599  Individual              ["AMP-AD"]   syn5592519   
..      ...          ...         ...                     ...          ...   
126     272          669  Consortium            ["MODEL-AD"]  syn50944316   
127     273          670  Consortium                     NaN  syn27207345   
128     274          672  Individual            ["Psych-AD"]  syn52160016   
129     275          675  Individual            ["M2OVE-AD"]  syn31511672   
130     276          677  Consortium            ["MODEL-AD"]  syn27207345   

                                          DataType_All  \
0                                  ["Gene Expression"]   
1                                  ["Gene Expression"]   
2                                       ["Proteomics"]   
3                                     ["Metabolomics"]   
4                                     ["Metabolomics"]   
..                                                 ...   
126  ["Gene Expression, Electrophysiology, Immunoas...   
127                                                NaN   
128                                ["Gene Expression"]   
129                                ["Gene Expression"]   
130                   ["Gene Expression, Immunoassay"]   

                                            Study_Name  \
0      The RNAseq Sample Swap Study (rnaSeqSampleSwap)   
1         The AD Cross Species Study (AD_CrossSpecies)   
2             The Adult Changes in Thought (ACT) Study   
3    The Alzheimer's Disease Metabolomics Consortiu...   
4    The Alzheimer's Disease Metabolomics Consortiu...   
..                                                 ...   
126                            The UCI_Bin1K358R Study   
127                                The UCI_ABCA7 Study   
128  The Mount Sinai Neuropsychiatric Symptoms in A...   
129  The Mayo Clinic Single Nucleus RNAseq Study (M...   
130                                The UCI_ABCA7 Study   

                                     Study_Description specimenType  \
0    This study provides transcriptomic data from a...    ["Brain"]   
1    This study provides transcriptomic data from a...    ["Brain"]   
2    This study provides proteomic data from the  d...    ["Brain"]   
3    This study provides metabolomics data generate...    ["Blood"]   
4    This study provides metabolomics data generate...    ["Blood"]   
..                                                 ...          ...   
126  This study provides bulk RNA-seq, immunohistoc...    ["Brain"]   
127                                                NaN    ["Brain"]   
128  This study provides single-nucleus RNAseq prof...    ["Brain"]   
129  This study provides single-nucleus transcripto...    ["Brain"]   
130                                                NaN          NaN   

                                            studyFocus  ... isModelSystem  \
0                          ["No Cognitive Impairment"]  ...         False   
1                                ["Alzheimer Disease"]  ...          True   
2                                ["Alzheimer Disease"]  ...         False   
3                                ["Alzheimer Disease"]  ...         False   
4                                ["Alzheimer Disease"]  ...         False   
..                                                 ...  ...           ...   
126  ["Alzheimer Disease", " Electrophysiology", "I...  ...          True   
127                              ["Alzheimer Disease"]  ...          True   
128  ["Alzheimer Disease", "Parkinson Disease", "Fr...  ...         False   
129                              ["Alzheimer Disease"]  ...    

In [47]:
study_citations = [Wiki(title=syn.get(study).name, 
                        owner='syn12666371',
                        markdown=f"", 
                        parentWikiId='628124') 
                   for study in portal_studies['Study'] if
                  study not in list(portal_studies['Study'].iloc[range(0,4)])]

#### Use `sqlite3` as a temporary fix until it's decided to version the Portal - Studies Table [section under development]

In [48]:
import sqlite3

#### Double-check length of `study_citations` to ensure all 131 rows are captured

In [49]:
len(study_citations)

127

#### There are 127 wiki objects in the list, and 4 already created, so the correct number of 127 + 4 = 131 wikis are created

#### Use `syn.store` to finalize creating the Citations wikis

In [51]:
"""for sc in study_citations:
    syn.store(sc)"""

#### Get Citation wiki ids

In [48]:
import requests

response = requests.get('https://repo-prod.prod.sagebase.org/repo/v1/entity/syn12666371/wikiheadertree2')


if response.status_code == 200:
    data = response.json()
    
else:
    print(f'Unsuccessful: {response.status_code}')
    print(response.text)

In [50]:
print(data)

{'totalNumberOfResults': 51, 'results': [{'id': '576279', 'title': ''}, {'id': '581939', 'title': 'About', 'parentId': '576279'}, {'id': '617505', 'title': 'Acknowledgement Statement Context', 'parentId': '576279'}, {'id': '602387', 'title': 'Acknowledgment Statements', 'parentId': '576279'}, {'id': '595383', 'title': 'Agora', 'parentId': '576279'}, {'id': '625598', 'title': 'Analysis Platforms', 'parentId': '576279'}, {'id': '628124', 'title': 'Citations', 'parentId': '576279'}, {'id': '600032', 'title': 'Contribute Data', 'parentId': '576279'}, {'id': '581937', 'title': 'Data Access', 'parentId': '576279'}, {'id': '605050', 'title': 'Frequently Asked Questions', 'parentId': '576279'}, {'id': '607137', 'title': 'Homepage- Resources Module', 'parentId': '576279'}, {'id': '608500', 'title': 'Old Synapse Portal Content', 'parentId': '576279'}, {'id': '595379', 'title': 'Study Details', 'parentId': '576279'}, {'id': '582408', 'title': "What's New", 'parentId': '576279'}, {'id': '585318', 

In [51]:
import requests

response = requests.get('https://repo-prod.prod.sagebase.org/repo/v1/entity/syn628124/wikiheadertree2')


if response.status_code == 200:
    data = response.json()
    
else:
    print(f'Unsuccessful: {response.status_code}')
    print(response.text)

Unsuccessful: 404
{"concreteType":"org.sagebionetworks.repo.model.ErrorResponse","reason":"A root wiki does not exist for ownerId: 628124 and ownerType: ENTITY"}
